Middleware-with-Tor/tor/tor/middlewares.py

```diff
@@ -0,0 +1,61 @@
+import sys
+import os
+from dotenv import load_dotenv
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+from stem.util.log import get_logger
+
+sys.path.append(os.path.abspath(".."))
+
+from settings import REQUESTS_PER_SAME_TOR_IDENTITY
+
+# load environment variables
+load_dotenv()
+
+# stop the logging in the terminal to not make it messy
+logger = get_logger()
+logger.propagate = False
+
+# Number of requests sent during the current tor identity
+NUM_SENT_REQUESTS = 0
```

Comment:
Bad practice to use a global variable when its not needed. This should be used as `ProxyMiddleware` instance attribute.

Reply:
Fixed.